XMonad/Util/ClickableWorkspaces.hs

```diff
@@ -0,0 +1,71 @@
+-------------------------------------------------------------------------------
+-- |
+-- Module      :  XMonad.Util.ClickableWorkspaces
+-- Copyright   :  (c) Geoff deRosenroll <geoffderosenroll@gmail.com>
+-- License     :  BSD3-style (see LICENSE)
+--
+-- Maintainer  :  Geoff deRosenroll <geoffderosenroll@gmail.com>
+-- Stability   :  unstable
+-- Portability :  unportable
+--
+-- Provides @clickablePP@, which when applied to the PP pretty-printer used by
+-- the "XMonad.Hooks.DynamicLog" hook, will make the workspace tags clickable in
+-- XMobar (for switching focus).
+--
+-----------------------------------------------------------------------------
+
+module XMonad.Util.ClickableWorkspaces (
+  -- * Usage
+  -- $usage
+  clickablePP
+  ) where
+
+import XMonad
+import XMonad.Util.WorkspaceCompare (getWsIndex)
+import XMonad.Hooks.DynamicLog (PP(..))
+
+-- $usage
+-- However you have set up your PP, apply @clickablePP@ to it, and bind the result
+-- to "XMonad.Hooks.DynamicLog"\'s dynamicLogWithPP like so:
+-- 
+-- > myPP xmprocs = xmobarPP { ... }
+-- > myLogHook xmprocs = (clickablePP . myPP) xmprocs >>= dynamicLogWithPP
+--
+-- * Requirements:
+--   * xdotool on system (in path)
+--   * use of UnsafeStdinReader in xmobarrc (rather than StdinReader)
+--   * Workspace switch bindings are expected to be "super + i",
+--     where i is the workspace index (beginning with 1), as in default bindings.
```

Comment:
Hardcoding this seems quite unfortunate to me. Personally I'd use `wmctrl -s`. That requires EWMH support in xmonad, but that's probably a safer assumption than "super + i" switching desktops?